In [26]:
import numpy as np
from scipy import stats

data = np.load("prices.npy")
numpy_arr = np.array(data)
print(numpy_arr.shape)
print(numpy_arr)

(11,)
[150 160 170 165 155 158 162 161 159 165 850]


In [3]:
Mean = np.mean(numpy_arr)
print("the mean with outliers is : ",Mean)

Mode = stats.mode(numpy_arr)
print("the mode with outliers is : ",Mode)

Median = np.median(numpy_arr)    
print("the median with outliers is : ",Median)

the mean with outliers is :  223.1818181818182
the mode with outliers is :  ModeResult(mode=np.int64(165), count=np.int64(2))
the median with outliers is :  161.0


In [4]:
numpy_arr[-1] = numpy_arr[0]
print(numpy_arr)

[150 160 170 165 155 158 162 161 159 165 150]


In [5]:
Mean = np.mean(numpy_arr)
print("the mean without outliers is : ",Mean)

Mode = stats.mode(numpy_arr)
print("the mode without outliers is : ",Mode)

Median = np.median(numpy_arr)    
print("the median without outliers is : ",Median)

the mean without outliers is :  159.54545454545453
the mode without outliers is :  ModeResult(mode=np.int64(150), count=np.int64(2))
the median without outliers is :  160.0


3. yes the outliear effect the average value by about 63
5. By removing the outlier (850) it lower down the mean value to 159.54 instead of 223.18
6. before removing the outliear the Median was good central tendancy while after removing the outlier
    mean performs better as central tendancy
    

In [7]:

### finding Zscore, std, mean and cv
from scipy.stats import zscore

In [ ]:
# std
z_score = zscore(numpy_arr)
print("z score is : ",z_score)

z score is :  [-1.61825982  0.07705999  1.7723798   0.9247199  -0.77059991 -0.26200397
  0.41612395  0.24659197 -0.09247199  0.9247199  -1.61825982]


In [23]:
import numpy as np
data =[10,20,30,40,50,60,70,80,90,100]
p25 = np.percentile(data,25)
p50 = np.percentile(data,50)
p75 = np.percentile(data,75)

print("25th percentile is : ",p25)
print("50th percentile is(median) : ",p50)
print("75th percentile is : ",p75)

25th percentile is :  32.5
50th percentile is(median) :  55.0
75th percentile is :  77.5


[ 90 180 270 360 450 540 630 720 810 900]


In [24]:
data = np.dot(np.array(data),3)
print(data)
p30 = np.percentile(data,30)
p70 = np.percentile(data,70)
p82 = np.percentile(data,82)

print("30th percentile is : ",p30)
print("70th percentile is(median) : ",p70)
print("82th percentile is : ",p82)

[ 30  60  90 120 150 180 210 240 270 300]
30th percentile is :  111.0
70th percentile is(median) :  219.0
82th percentile is :  251.4


Practice Task
1. Download prices.npy form google class room
2. load using numpy
3. calculate q1, q2, q3
4. find the IQr
5. Determine if any values are outliers (use IQR method)

In [28]:
data1 = np.load("prices.npy")
numpy_arr1 = np.array(data)
print(numpy_arr1.shape)
print(numpy_arr1)

(11,)
[150 160 170 165 155 158 162 161 159 165 850]


In [30]:
## calculate Q1, Q2, Q3
Q1 = np.percentile(numpy_arr1,25)
Q2 = np.percentile(numpy_arr1,50)
Q3 = np.percentile(numpy_arr1,75)

print("Q1 is : ",Q1)
print("Q2 is(median) : ",Q2)
print("Q3 is : ",Q3)

Q1 is :  158.5
Q2 is(median) :  161.0
Q3 is :  165.0


In [31]:
# now finding IQR
IQR = Q3 - Q1
print("IQR is : ",IQR)

IQR is :  6.5


In [37]:
## determine whether any values are outliers (use IQR method)

## if a value is greater than Q3 + 1.5 * IQR or less than Q1 - 1.5 * IQR then it is outlier
for i in range(len(numpy_arr1)):
    if numpy_arr1[i] > Q3 + 1.5 * IQR or numpy_arr1[i] < Q1 - 1.5 * IQR:
        print("outlier")
        print(numpy_arr1[i])
    else:
       pass # print("not outlier")

outlier
850
